services/worker-manager/src/api.js

```diff
@@ -474,6 +474,67 @@ builder.declare({
   return res.reply(worker.serializable());
 });
 
+builder.declare({
+  method: 'post',
+  route: '/workers/:workerPoolId:/:workerGroup/:workerId',
+  name: 'updateWorker',
+  title: 'Update an existing Worker',
+  category: 'Workers',
+  stability: APIBuilder.stability.stable,
+  input: 'create-worker-request.yml',
+  output: 'worker-full.yml',
+  // note that this pattern relies on workerGroup and workerId not containing `/`
+  scopes: 'worker-manager:update-worker:<workerPoolId>/<workerGroup>/<workerId>',
+  cleanPayload: cleanCreatePayload,
+  description: [
+    'Update an existing worker.  The precise behavior of this method depends',
+    'on the provider implementing the given worker pool.  Some providers',
+    'do not support updating workers at all, and will return a 400 error.',
+  ].join('\n'),
+}, async function(req, res) {
+  // TODO: test
+  const { workerPoolId, workerGroup, workerId } = req.params;
```

Comment:
Should there be a call somewhere to `req.authorize` to check scope permissions?

Reply:
See [the tc-lib-api docs](https://github.com/taskcluster/taskcluster/tree/main/libraries/api#scopes) -- `req.authorize` is only reqiured when the scope expression requires information that's not in `req.params`.  And if it's missed, then the API method fails very loudly.